front/src/common/IntervalsDataViz/Scales.tsx

```diff
@@ -1,54 +1,123 @@
 import React, { useEffect, useState } from 'react';
 import cx from 'classnames';
 
-import { roundNumber, shortNumber } from './utils';
+import { isRoundKm, roundNumber, shortNumber } from './utils';
+
+/**
+ * Given a value, return the absolute position (in px) corresponding to it on the graph.
+ * If the value is not in the interval [begin, end], return 0.
+ */
+const getPositionFromValue = (
+  value: number,
+  begin: number,
+  end: number,
+  graphWidth: number,
+  leftPadding: number
+) =>
+  value > begin && value < end ? ((value - begin) * graphWidth) / (end - begin) + leftPadding : 0;
 
 export const ResizingScale = ({
   begin,
   end,
-  className,
+  wrapper,
 }: {
   begin: number;
   end: number;
-  className?: string;
+  wrapper: HTMLElement;
 }) => {
-  const [ticksCount, setTicksCount] = useState<number>(10);
+  const [ticks, setTicks] = useState<{ position: number; value: number; isExtremity: boolean }[]>(
+    []
+  );
+
+  const computeTicks = (_begin: number, _end: number, _wrapper: HTMLElement) => {
+    const graphWidth = wrapper.offsetWidth;
+    const leftPadding = wrapper.offsetLeft;
+
+    // compute the number of ticks, knowing that space between 2 ticks should be 200px
+    const ticksCount = Math.round(graphWidth / 200);
+
+    // compute the default step between 2 values (we want to display only value in km)
+    const step = roundNumber((_end - _begin) / ticksCount / 1000) * 1000;
+
+    // compute the ticks (value and position)
+    const newtTicks = [{ value: _begin, position: leftPadding, isExtremity: true }];
+    if (step > 0) {
+      // find the first tick value (first integer above begin (in km) or begin if it is x km)
+      let index = isRoundKm(begin) ? 1 : 0;
+      const firstTick = isRoundKm(begin) ? _begin : roundNumber(begin / 1000, true) * 1000;
 
-  const inf = roundNumber(begin, true);
-  const sup = roundNumber(end, false);
-  const step = roundNumber((end - begin) / ticksCount / 2, true);
+      // check the second tick is not too close to the first one
+      const secondTickPosition = getPositionFromValue(
+        step * index + firstTick,
+        _begin,
+        _end,
+        graphWidth,
+        leftPadding
+      );
+      if (secondTickPosition - leftPadding < 20) {
```

Comment:
put this twenty as a constant for start pls

Reply:
Done, used MIN_DISTANCE_TO_EXTREMITY